firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -39,26 +43,38 @@
 import com.google.android.gms.tasks.Tasks;
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.installations.FirebaseInstallationsApi;
+import com.google.firebase.installations.InstallationTokenResult;
+import java.net.ProtocolException;
 import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class FirebaseAppDistribution implements Application.ActivityLifecycleCallbacks {
 
   private final FirebaseApp firebaseApp;
   private final FirebaseInstallationsApi firebaseInstallationsApi;
+  private final FirebaseAppDistributionTesterApiClient firebaseAppDistributionTesterApiClient;
   private static final String TAG = "FirebaseAppDistribution";
   private Activity currentActivity;
-  @VisibleForTesting private boolean currentlySigningIn = false;
+  private boolean currentlySigningIn = false;
   private TaskCompletionSource<Void> signInTaskCompletionSource = null;
   private CancellationTokenSource signInCancellationSource;
 
+  private TaskCompletionSource<AppDistributionRelease> checkForUpdateTaskCompletionSource = null;
+  private CancellationTokenSource checkForUpdateCancellationSource;
+  private Executor checkForUpdateExecutor;
+
   /** Constructor for FirebaseAppDistribution */
   public FirebaseAppDistribution(
       @NonNull FirebaseApp firebaseApp,
-      @NonNull FirebaseInstallationsApi firebaseInstallationsApi) {
+      @NonNull FirebaseInstallationsApi firebaseInstallationsApi,
+      @NonNull FirebaseAppDistributionTesterApiClient firebaseAppDistributionTesterApiClient) {
     this.firebaseApp = firebaseApp;
     this.firebaseInstallationsApi = firebaseInstallationsApi;
+    this.firebaseAppDistributionTesterApiClient = firebaseAppDistributionTesterApiClient;
+    this.checkForUpdateExecutor = Executors.newFixedThreadPool(4);
```

Comment:
Pull `4` here into a named constant `UPDATE_THREAD_POOL_SIZE`, consider if 4 is the right number of threads